src/switchx/Switchx.hx

```diff
@@ -216,15 +221,22 @@ class Switchx {
       root.delete();
   }
       
-  public function download(version:ResolvedVersion, options:{ force: Bool }):Promise<Bool>
+  public function download(version:ResolvedVersion, options:{ force: Bool }):Promise<Bool> {
+    
+    inline function downloadArchive(url, peel, into)
```

Comment:
Fun fact: [`Download.archive`](https://github.com/lix-pm/switchx/blob/master/src/switchx/Download.hx#L36) does pretty much the same thing as `downloadArchive` :P